README.md

```diff
@@ -1 +1,101 @@
-unity-hierarchy-behaviour
+# unity-hierarchy-behaviour
+
+# What is it?
+Its a collection of GameObject extension methods to allow for runtime instantiation of MonoBehaviours that include an Initialize method that takes type-safe arguments.
+
+## What are the Core Features?
+- CreateChild (New, Resources, Loaded or Instantiated)
+- ReplaceChild (New, Resources, Loaded or Instantiated)
+
+## Why does it exist?
+The ability to easily add child game objects with behaviours, initialize them and pass in type-safe args
```

Comment:
I'd like to drive home the benefits of having this ability as well here. Mainly why we put things into a hierarchy.
- Control the flow of data,
- Divide responsibility of components
- Visualize that responsibility
- Control lifecycles